src/calculate_dependencies.py

```diff
@@ -1,36 +1,167 @@
+import os
+import re
+
+import numpy as np
 import pandas as pd
 import datetime as dt
 
+from generator import GeneratorFactory, GeneratorType
+
```

Comment:
Overall good, check again for redundancy of comments and separation of concerns.